test/browser/redux/actions/async.test.js

```diff
@@ -1675,14 +1891,204 @@ describe('Asynchronous Actions', () => {
         ];
         asyncActions.__Rewire__('services', {
           localStore: {
-            getItem: () => null
+            getItem: () => null,
+            removeItem: (item) => null
           }
         });
         const store = mockStore({}, expectedActions, done);
         store.dispatch(asyncActions.retrieveTargetsFromStorage());
       });
     });
 
+    describe('no targets retrieved from local storage, targets exist in state but no user targeted for upload by default', () => {
+      it('should dispatch RETRIEVING_USERS_TARGETS, SET_PAGE (redirect to settings page for user selection)', (done) => {
+        const expectedActions = [
+          {
+            type: actionTypes.RETRIEVING_USERS_TARGETS,
+            meta: {source: actionSources[actionTypes.RETRIEVING_USERS_TARGETS]}
+          },
+          {
+            type: actionTypes.SET_PAGE,
+            payload: {page: pages.SETTINGS},
+            meta: {source: actionSources[actionTypes.SET_PAGE]}
+          }
+        ];
+        asyncActions.__Rewire__('services', {
+          localStore: {
+            getItem: () => null,
+            removeItem: (item) => null
+          }
+        });
+        const state = {
+          allUsers: {
+            ghi789: {},
+            abc123: {},
+            def456: {},
+          },
+          loggedInUser: 'ghi789',
+          targetsForUpload: ['abc123', 'def456'],
+          uploadTargetUser: null,
+          targetDevices: {
+            'abc123': ['carelink']
+          }
+        };
+        const store = mockStore(state, expectedActions, done);
+        store.dispatch(asyncActions.retrieveTargetsFromStorage());
+      });
+    });
+
+    describe('no targets retrieved from local storage, targets exist in state but user targeted for upload is missing timezone', () => {
```

Comment:
Outdated now that timezone selection is always on main page, no? At least update the description.